frameworks/OCaml/webmachine/src/src/bin/tfb.ml

```diff
@@ -138,7 +139,46 @@ class queries =
       Wm.continue (`String (Lib.Db_j.string_of_queries json)) rd
   end
 
+let dump_lwt () =
+  let options =
+    [
+      ("fd_passing", `fd_passing);
+      ("fdatasync", `fdatasync);
+      ("get_affinity", `get_affinity);
+      ("get_cpu", `get_cpu);
+      ("get_credentials", `get_credentials);
+      ("libev", `libev);
+      ("madvise", `madvise);
+      ("mincore", `mincore);
+      ("recv_msg", `recv_msg);
+      ("send_msg", `send_msg);
+      ("set_affinity", `set_affinity);
+      ("wait4", `wait4);
+    ]
+  in
+  ignore @@ Lwt_io.eprintf "Lwt:\n";
```

Comment:
We probably shouldn't ignore the return value (promise) of all the async Lwt functions.
Alternatively, you can use regular Printf when you're not serving requests yet.

Reply:
I agree I just don't know how to make it work without either ignoring the promise or using the regular Printf. My Lwt foo is pretty weak so any pointers on how to make this work with `Lwt_io` without `ignore` would be appreciated.

Reply:
Lwt exposes a monadic API (monadic just means that you chain things together using the `bind`, aka. `>>=` or `>|=` operators) to allow you to do just that.

Currently, you're starting up async computations in a fire and forget fashion instead of executing them sequentially. If you're familiar with Javascript promises, you can think of `>>=` and `>|=` as `.then()`. You can look at my code in `httpaf_unix.ml` for an example of using the monadic API. There's a lot of information encoded in the type signatures of the Lwt monadic functions, do not hesitate to scrutinize them, as well as the type signatures of `>>=`, `>|=` and `Lwt_main.run`.